lib/internal/modules/esm/initialize_import_meta.js

```diff
@@ -47,6 +53,26 @@ function initializeImportMeta(meta, context, loader) {
   return meta;
 }
 
+/**
+ * Inspect the module URL and convert it to `__dirname` and `__filename`
+ * references if it does not point to a network module.
+ * @param {string} url
+ * @returns {{__dirname?: string, __filename?: string}}
+ */
+function resolveModuleMeta(url) {
+  if (url.slice(0, 7) !== 'file://') {
+    // These only make sense for locally loaded modules,
+    // i.e. network modules are not supported.
+    return { __dirname: undefined, __filename: undefined };
+  }
+
+  const filePath = fileURLToPath(url);
+  return {
+    __dirname: dirname(filePath),
+    __filename: filePath,
+  };
```

Comment:
```suggestion
  let filePath;
  let dirPath;
  return {
    get __dirname() {
       return dirname(filePath ??= fileURLToPath(url))
    },
    get __filename() {
       return dirPath ??= dirname(filePath ??= fileURLToPath(url))
    }
  };
```

Reply:
came from comment https://github.com/nodejs/node/pull/48740/files#r1260338672

Reply:
Thanks. That makes sense. The `__filename` getter isn't quite correct, so I made some adjustments in https://github.com/nodejs/node/pull/48740/commits/aa11c249289b4e02e9dc649d42c6c16f296a952e